### DISCUSSION TITLE
A periodic computational domain with EB

### DISCUSSION THREAD

**Discussion by ZhaofanZhu** [AMReX-Combustion/PeleLMeX#327] [Q&A]
I want to simulate a cubic domain with periodic boundary condition in x and z directions. As shown in the figure below.
There is an EB plane using 'AMReX_EB2_IF_Plane.H' at y=0.

The simulation is fine when I set  periodic boundary condition in z direction and outflow boundary condition in x direction. Once I set both the x and z directions to periodic boundary condition, it's error with:
```
==============================================================================
 Est. time step - Conv: 1e+199, divu: 1e+20
 Initial dt: 1e+17

 Doing initial projection(s) 

 Initial velocity projection:   U: 0  V: 0  W: 0
amrex::amrex::Abort::0::MLMG failed. !!!
SIGABRT
amrex::Abort::126::MLMG failed. !!!
SIGABRT
amrex::Abort::64::MLMG failed. !!!
SIGABRT
amrex::Abort::189::MLMG failed. !!!
```

I don't know why this is happening, looking forward to your reply.

![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/144213811/956133b1-8c39-4d43-a948-6c5535c514a7)

**Comment by ZhaofanZhu** [comment]
By the way, the simulation is also fine with periodic x and outflow z.

**Comment by ZhaofanZhu** [comment]
@baperry2 Hi Bruce, doesn't the PeleLMeX support the periodicity of EB in two direactions? Do you have any advice? Thank you in advance!

**Comment by baperry2** [comment]
I'm not aware of a reason why this wouldn't work (provided the EB surface matches across the periodic boundary). But this type of configuration with an EB intersecting periodic boundaries is not in any of our tests and I'm not aware of it being run before. So it's possible that I'm overlooking something, or that there are some bugs to shake loose.

For your case as described, as the EB plane is coincident to the ylo boundary, the clear solution is just to remove EB and set the appropriate BCs on the ylo domain boundary. But, assuming you are working toward a case where the EB will be necessary, it may be beneficial to move the EB plane to occur a few grid cells away from the boundary rather than being coincident with it. Also, to debug your case, you should turn up diagnostic output with `peleLM.v = 3`, `nodal_proj.verbose = 3`, `mac_proj.verbose = 3`, and `diffusion.verbose = 3` to try to determine which linear solve is failing and how.

**Comment by esclapez** [comment]
@ZhaofanZhu Just catching up with this. I agree with Bruce, nothing wrong with EB + periodic boundary. [This channel flow case](https://amrex-combustion.github.io/PeleLMeX/manual/html/Validation.html#channel-flow-using-eb) is performed in a periodic domain with EB. But it looks like your initial solution has a zero-velocity and you do not have an inflow or a background pressure gradient, so basically nothing is driving your flow. In this case, the time step size estimate return huge value which might be the cause of the projection failure. Can you try specifying an inital time step size:
```
amr.init_dt = 1e-6
```
You will still need a pressure gradient to drive the flow since your initial solution has a zero velocity:
```
peleLM.gradP0 = 0.0 0.0 -10.0
```
for instance to drive the flow in the +z direction.
